### PR TITLE
S3 import

### PR DESCRIPTION
**Pull request by scottaubrey**
fixes https://github.com/elifesciences/enhanced-preprints-issues/issues/502

When `S3_BUCKET` is set to anything truthy (alongside other s3 config env vars) we switch to using that as an import mechanism

built in support for file-based AWS session-token (for k8s)

### CODE REVIEW

**Review by E-Kemp — APPROVED** [review]
LGTM

File: docker-compose.yaml [hunk 1]
```diff
@@ -31,6 +31,31 @@ services:
     ports:
       - "8081:8081"
 
+  minio:
```
Comment (nlisgo) [inline docker-compose.yaml:34]: How do we ensure that this container is not part of the production build? Does it belong in a separate `docker-compose.ci.yaml`?

**Review by will-byrne — APPROVED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (will-byrne) [reply docker-compose.yaml:34]: I believe we are not using docker-compose for prod, just the image loaded in via kubernetes

Comment (scottaubrey) [reply docker-compose.yaml:34]: > I believe we are not using docker-compose for prod, just the image loaded in via kubernetes

Exactly. docker compose is not used for either image building or production runs so there's no worry in it being here. It's used for development only, and minio is also there for development and testing only, so belongs in the `docker-compose.yaml` (or maybe `docker-compose.override.yaml` if we were trying to do something only in dev and not in tests)

Comment (will-byrne) [reply docker-compose.yaml:34]: > (or maybe docker-compose.override.yaml if we were trying to do something only in dev and not in tests)

We aren't using docker-compose for the tests either at the moment, the integration tests run with mocked external services using supertest

**Review by nlisgo — APPROVED** [review]
LGTM

**Pull request merged**